File: breakfast-drone/drone.js
// given an array if ids, which contain many duplicate integers and one unique integer, find the unique integer

function uniqueID(ids) {
  let idsOccurence = {};

  for (let i = 0; i < ids.length; i++) {
    if (Object.values(idsOccurence).indexOf(ids[i]) > -1) {
      idsOccurence[ids[i]] += 1;
    } else {
      idsOccurence[ids[i]] = 1;
    }
  }
}

uniqueID([1, 1, 2, 2, 3, 4, 4, 4, 5, 5, 5, 6, 6, 6, 7, 7, 8, 8, 8]);

// iterate over the array
// create an object for each element with the number as the key and the times it appears as the value
// the number which has a value of one is the uniqueID

File: breakfast-drone/drone.rb
require 'pry'

def uniqueID(ids)
  idsOccurence = {}

  ids.each do |id|
    if idsOccurence.include?(id)
      idsOccurence[id] += 1
    else
      idsOccurence[id] = 1
    end
  end

  idsOccurence.each do |id, occurence|
    if occurence == 1
      puts id
    end
  end
end

uniqueID([1, 1, 2, 2, 3, 4, 4, 4, 5, 5, 5, 6, 6, 6, 7, 7, 8, 8, 8]); # should return 3
